lib/lockfile.js

```diff
@@ -93,33 +93,17 @@ function updateLock(file, options) {
 
     lock.updateDelay = lock.updateDelay || options.update;
     lock.updateTimeout = setTimeout(() => {
-        const mtime = Date.now() / 1000;
-
         lock.updateTimeout = null;
 
-        options.fs.utimes(getLockFile(file, options), mtime, mtime, (err) => {
-            // Ignore if the lock was released
-            if (lock.released) {
-                return;
-            }
-
-            // Verify if we are within the stale threshold
-            if (lock.lastUpdate <= Date.now() - options.stale && lock.lastUpdate > Date.now() - (options.stale * 2)) {
-                const err = Object.assign(
-                    new Error(lock.updateError || 'Unable to update lock within the stale threshold'),
-                    { code: 'ECOMPROMISED' }
-                );
-
-                return setLockAsCompromised(file, lock, err);
-            }
-
-            // If the file is older than (stale * 2), we assume the clock is moved manually,
-            // which we consider a valid case
+        // Check if mtime is still ours if it is we can still recover from a system sleep or a busy event loop
+        options.fs.stat(getLockFile(file, options), (err, stat) => {
+            const isMtimeOurs = lock.mtime.getTime() === (stat ? stat.mtime.getTime() : null);
```

Comment:
Nit: personally I'd move this after the error handling block so you don't need to use a ternary 😄